Tasks/ExtractFiles/extractFiles.ts

```diff
@@ -0,0 +1,319 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+
+var pattern: string = tl.getInput('pattern', true).trim();
+
+var destinationFolder :string  = tl.getPathInput('destinationFolder', true, false).trim();
+
+var cleanDestinationFolder: boolean = tl.getBoolInput('cleanDestinationFolder', false);
+
+var repoRoot : string = path.resolve(tl.getVariable('build.sourcesDirectory') || '');
+tl.debug('repoRoot: ' + repoRoot);
+
+var win = os.type().match(/^Win/);
+tl.debug('win: ' + win);
+
+// extractors
+var tarLocation : string;
+var _sevenZipLocation : string; // lazily loaded, use accessor method
+
+function getSevenZipLocation() {
+    if (_sevenZipLocation == undefined) {
+
+        if (win) {
+            // http://www.7-zip.org/
+            _sevenZipLocation = '7zip/7z.exe';
+        } else {
+            // https://sourceforge.net/projects/p7zip/
+            _sevenZipLocation = 'p7zip/7z';
+
+        }
+    }
+    return _sevenZipLocation;
+}
+
+function findFiles(fullSearchString: string): string[] {
+    var patterns: string[] = splitPattern(fullSearchString);
```

Comment:
We shouldn't split on commas, task should have multiline input control.  See copy files https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/CopyFiles/copyfiles.ts